app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksListContentHeader/MocksListContentHeader.tsx

```diff
@@ -1,55 +1,113 @@
-import React from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import { useSelector } from "react-redux";
 import { CloudUploadOutlined, PlusOutlined } from "@ant-design/icons";
-import { ButtonProps } from "antd";
+import { Badge, ButtonProps } from "antd";
 import { AuthConfirmationPopover } from "components/hoc/auth/AuthConfirmationPopover";
-import { ContentListHeader } from "componentsV2/ContentList";
-import { MockType } from "components/features/mocksV2/types";
+import { ContentListHeader, ContentListHeaderProps } from "componentsV2/ContentList";
+import {
+  MockType,
+  MockTableHeaderFilter,
+  RQMockMetadataSchema,
+  MockListSource,
+} from "components/features/mocksV2/types";
 import { RQButton } from "lib/design-system/components";
 import { SOURCE } from "modules/analytics/events/common/constants";
 import { getUserAuthDetails } from "store/selectors";
+import { MdOutlineCreateNewFolder } from "@react-icons/all-files/md/MdOutlineCreateNewFolder";
+import { MdOutlineStarOutline } from "@react-icons/all-files/md/MdOutlineStarOutline";
+import { isRecordMock } from "../MocksTable/utils";
+import { useMocksActionContext } from "features/mocks/contexts/actions";
+import { useLocation } from "react-router-dom";
+import PATHS from "config/constants/sub/paths";
+import { getQuickFilteredRecords } from "./utils";
 
 interface Props {
-  mockType?: string;
-  handleUploadAction?: () => void;
-  handleCreateNew: () => void;
+  source?: MockListSource;
+  mockType?: MockType;
+  records?: RQMockMetadataSchema[];
   searchValue?: string;
   setSearchValue?: (s: string) => void;
+  filter?: MockTableHeaderFilter;
+  setFilter?: (filter: MockTableHeaderFilter) => void;
+  setFilteredMocks?: (mocks: RQMockMetadataSchema[]) => void;
+  handleCreateNewMockFromPickerModal?: () => void;
 }
 
 export const MocksListContentHeader: React.FC<Props> = ({
+  source,
+  records,
   mockType,
-  handleCreateNew,
-  handleUploadAction,
+  filter,
   searchValue,
+  setFilter,
   setSearchValue = () => {},
+  setFilteredMocks = () => {},
+  handleCreateNewMockFromPickerModal = () => {},
 }) => {
+  const [mocks, setMocks] = useState<RQMockMetadataSchema[]>([]);
+
   const user = useSelector(getUserAuthDetails);
+  const { pathname } = useLocation();
+  const { createNewCollectionAction, mockUploaderModalAction, createNewMock } = useMocksActionContext() ?? {};
+  const isRuleEditor = pathname.includes(PATHS.RULE_EDITOR.RELATIVE);
+
+  useEffect(() => {
+    const filteredRecords = getQuickFilteredRecords(records, filter);
+    setMocks(filteredRecords);
+    setFilteredMocks(filteredRecords);
+  }, [filter, records]);
```

Comment:
Why are we doing filtering here now. Shouldn't this be in parent component only?

Reply:
Also this is not working along with search logic as both are trying to modify same filteredRecords